test/system/src/test/java/io/pravega/test/system/MultiControllerTest.java

```diff
@@ -176,8 +178,7 @@ private void withControllerURIDiscover() throws ExecutionException, InterruptedE
     }
 
     private CompletableFuture<Boolean> createScope(String scopeName, URI controllerURI) {
```

Comment:
Here is how I think you should refactor this method:

```
private Boolean createScope(String scopeName, URI controllerURI) {
      @Cleanup
       final ControllerImpl controllerClient = new ControllerImpl(ControllerImplConfig.builder()
                 .clientConfig(ClientConfig.builder()
                 .controllerURI(controllerURI)
                 .build())
                 .build(), EXECUTOR_SERVICE);
         return controllerClient.createScope(scopeName).get();
}

```

Of course, you'd need to remove the `get()` calls in the test code, but the functionality does not change as far as I can tell. And, given that this is a private method, there is no concern around methods outside this class.

Reply:
I have updated the PR, please check if this works.